docs/standard/base-types/backtracking-in-regular-expressions.md

```diff
@@ -198,6 +206,16 @@ If you do not set a time-out value explicitly, the default time-out value is det
 |`[A-Z]\w*`|Match an alphabetical character followed by zero or more word characters.|
 |`$`|End the match at the end of the input string.|
 
+### General Performance Considerations
+
+The following suggestions are not specifically to prevent excessive backtracking, but may help increase the performance of your regular expression:
+
+1. If a pattern will be heavily used, precompile it. The best way to do this is to use the [regular expression source generator](regular-expression-source-generators.md) to precompile it. If the source generator is not available for your app, use the <xref:System.Text.RegularExpressions.RegexOptions.Compiled?displayProperty=nameWithType> option.
+
+1. If matches are known to always start beyond a certain offset into the pattern, consider passing the offset to the regular expression engine. For example, using the overload <xref:System.Text.RegularExpressions.Regex.Match%28System.String%2CSystem.Int32%29?displayProperty=nameWithType>. This will reduce the amount of the text the engine needs to consider.
+
+1. If matches are typically found near the end of a very large input, <xref:System.Text.RegularExpressions.RegexOptions.RightToLeft?displayProperty=nameWithType> might find it more quickly. However, use this with care. The regular expression engines are most heavily optimized for left to right operation, so you should test to ensure this is an improvement. It may be better to pass in the offset instead.
```

Comment:
@stephentoub I'm hesitant about including this. I'm not sure we even measure the perf of RTL.